File: 4119-minimum-distance-between-three-equal-elements-ii/minimum-distance-between-three-equal-elements-ii.py
import math
from collections import defaultdict
class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        pos = defaultdict(list)
        for i, num in enumerate(nums):
            pos[num].append(i)
        ans = math.inf
        for indices in pos.values():
            if len(indices) >= 3:
                for i in range(len(indices) - 2):
                    ans = min(ans, 2 * (indices[i + 2] - indices[i]))

        return -1 if ans == math.inf else ans

File: 4119-minimum-distance-between-three-equal-elements-ii/minimum-distance-between-three-equal-elements-ii.js
var minimumDistance = function (nums) {
    const pos = new Map()
    for (let i = 0; i < nums.length; i++) {
        if (!pos.has(nums[i])) {
            pos.set(nums[i], [])
        }
        pos.get(nums[i]).push(i)
    }
    let ans = Infinity
    for (const [val, indices] of pos.entries()) {
        if (indices.length >= 3) {
            for (let i = 0; i <= indices.length - 3; i++) {
                ans = Math.min(ans, 2 * (indices[i + 2] - indices[i]))
            }
        }
    }
    return ans === Infinity ? -1 : ans
};